include/openssl/ocsp.h

```diff
@@ -194,6 +194,11 @@ OPENSSL_EXPORT int OCSP_request_add1_nonce(OCSP_REQUEST *req,
 //  but aren't equal.
 OPENSSL_EXPORT int OCSP_check_nonce(OCSP_REQUEST *req, OCSP_BASICRESP *bs);
 
+// OCSP_copy_nonce copies the nonce value (if any) from |req| to |resp|. Returns
+// 1 on success and 0 on failure. If the optional nonce value does not exist in
+// |req|, we return 2 instead.
+OPENSSL_EXPORT int OCSP_copy_nonce(OCSP_BASICRESP *resp, OCSP_REQUEST *req);
```

Comment:
nit: should we document that the nonce extension is prepended?

Reply:
Ah yes we should

Reply:
The more I think about it, I think the correct behavior should be replacing any existing nonce in `resp` instead of prepending/appending it to the list of extensions. OpenSSL appends the nonce to the list, but it doesn't really make sense for a `resp` to have two nonces.
My initial concern was that someone may have noticed OpenSSL's double nonce behavior and they're removing the old extra nonce themselves after the call to `OCSP_copy_nonce`. This might cause our enforced singular nonce to be removed unintentionally. But it's an edge case trying to get around OpenSSL's buggy behavior, enforcing the right behavior might be the better call here.

Reply:
After discussing with @WillChilds-Klein, ultimately decided to align with OpenSSL's implementation. There's not too much benefit in misaligning here to be fair and creates more unanticipated behavioral changes.